LibreSSL uses a synthetic version in order to force consumers to check
individual features instead but API isn't compatible with OpenSSL 1.1.x.

https://github.com/libevent/libevent/commit/d057c45e8f48

Index: openssl-compat.h
--- openssl-compat.h.orig
+++ openssl-compat.h
@@ -1,9 +1,9 @@
 #ifndef OPENSSL_COMPAT_H
 #define OPENSSL_COMPAT_H
 
-#if OPENSSL_VERSION_NUMBER < 0x10100000L
+#if (OPENSSL_VERSION_NUMBER < 0x10100000L) || defined(LIBRESSL_VERSION_NUMBER)
 
-static inline BIO_METHOD *BIO_meth_new(int type, const char *name)
+inline BIO_METHOD *BIO_meth_new(int type, const char *name)
 {
 	BIO_METHOD *biom = calloc(1, sizeof(BIO_METHOD));
 
@@ -30,6 +30,6 @@ static inline BIO_METHOD *BIO_meth_new(int type, const
 
 #define TLS_method SSLv23_method
 
-#endif /* OPENSSL_VERSION_NUMBER < 0x10100000L */
+#endif /* (OPENSSL_VERSION_NUMBER < 0x10100000L) || defined(LIBRESSL_VERSION_NUMBER) */
 
 #endif /* OPENSSL_COMPAT_H */
